modules/index/elastic-client/src/test/scala/ch/epfl/bluebrain/nexus/commons/es/client/ElasticServer.scala

```diff
@@ -0,0 +1,79 @@
+package ch.epfl.bluebrain.nexus.commons.es.client
+
+import java.nio.file.Files
+import java.util.Arrays._
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.stream.ActorMaterializer
+import akka.testkit.TestKit
+import ch.epfl.bluebrain.nexus.commons.es.client.ElasticServer.MyNode
+import ch.epfl.bluebrain.nexus.commons.test.Randomness
+import org.apache.commons.io.FileUtils
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.node.Node
+import org.elasticsearch.plugins.Plugin
+import org.elasticsearch.transport.Netty3Plugin
+import org.scalatest.{BeforeAndAfterAll, WordSpecLike}
+
+abstract class ElasticServer
+    extends TestKit(ActorSystem("ElasticServer"))
+    with WordSpecLike
+    with BeforeAndAfterAll
+    with Randomness {
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    startElastic()
+  }
+
+  override protected def afterAll(): Unit = {
+    stopElastic()
+    super.afterAll()
+    TestKit.shutdownActorSystem(system)
+  }
+
+  val esUri       = Uri(s"http://localhost:9200")
```

Comment:
it would be very nice to determine a free port to use for binding elastic search since on a build machine may have multiple executors which could make ports collide.

Reply:
I tried that and for some reason It wasn't always working. I will give a second look at it